module/observations/helper.js

```diff
@@ -2819,20 +2823,53 @@ module.exports = class ObservationsHelper {
           let topLevelEntityId = bodyData[topLevelEntityType];
 
 
-          let roles = bodyData.role.split(",");
+          let tenantDetails = await userService.fetchPublicTenantDetails(tenantData.tenantId);
+          if (!tenantDetails.data || !tenantDetails.data.meta || tenantDetails.success !== true) {
```

Comment:
@borkarsaish65  we can optimise this check
if (!tenantDetails.success || !tenantDetails?.data?.meta ) {

Reply:
done